xarray/tests/test_units.py

```diff
@@ -355,7 +365,7 @@ def test_apply_ufunc_dataarray(dtype):
     expected = attach_units(func(strip_units(data_array)), extract_units(data_array))
     actual = func(data_array)
 
-    assert_equal_with_units(expected, actual)
+    assert_identical(expected, actual)
```

Comment:
So `assert_identical` checks units too? Let's add that to the docstring.

Reply:
well, `identical` seems to check the units, so `assert_identical` does, too. I didn't change anything, using a `pint` version with the changes from hgrecco/pint#905 was enough.